src/main/java/org/opensearch/timeseries/NodeStateManager.java

```diff
@@ -235,68 +157,140 @@ public boolean isMuted(String nodeId, String detectorId) {
+                }
+
+                function.accept(Optional.of(config));
+            } catch (Exception e) {
+                String message = "Failed to parse config " + configId;
+                LOG.error(message, e);
+                listener.onFailure(new OpenSearchStatusException(message, RestStatus.INTERNAL_SERVER_ERROR));
+            }
+        }, exception -> {
+            LOG.error("Failed to get config " + configId, exception);
+            listener.onFailure(exception);
+        }));
     }
 
-    /**
-     * Get last error of a detector
-     * @param adID detector id
-     * @return last error for the detector
-     */
-    public String getLastDetectionError(String adID) {
-        return Optional.ofNullable(states.get(adID)).flatMap(state -> state.getLastDetectionError()).orElse(NO_ERROR);
+    public void getConfig(String configID, AnalysisType context, ActionListener<Optional<? extends Config>> listener) {
+        NodeState state = states.get(configID);
+        if (state != null && state.getConfigDef() != null) {
+            listener.onResponse(Optional.of(state.getConfigDef()));
+        } else {
+            GetRequest request = new GetRequest(CommonName.CONFIG_INDEX, configID);
+            BiCheckedFunction<XContentParser, String, ? extends Config, IOException> configParser = context == AnalysisType.AD
+                ? AnomalyDetector::parse
+                : Forecaster::parse;
+            clientUtil.<GetRequest, GetResponse>asyncRequest(request, client::get, onGetConfigResponse(configID, configParser, listener));
+        }
     }
 
-    /**
-     * Set last detection error of a detector
-     * @param adID detector id
-     * @param error error, can be null
-     */
-    public void setLastDetectionError(String adID, String error) {
-        NodeState state = states.computeIfAbsent(adID, id -> new NodeState(id, clock));
-        state.setLastDetectionError(error);
+    private ActionListener<GetResponse> onGetConfigResponse(
+        String configID,
+        BiCheckedFunction<XContentParser, String, ? extends Config, IOException> configParser,
+        ActionListener<Optional<? extends Config>> listener
+    ) {
+        return ActionListener.wrap(response -> {
+            if (response == null || !response.isExists()) {
+                listener.onResponse(Optional.empty());
+                return;
+            }
+
+            String xc = response.getSourceAsString();
+            LOG.debug("Fetched config: {}", xc);
+
+            try (
+                XContentParser parser = XContentType.JSON.xContent().createParser(xContentRegistry, LoggingDeprecationHandler.INSTANCE, xc)
+            ) {
+                ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+                Config config = configParser.apply(parser, response.getId());
+
+                // end execution if all features are disabled
+                if (config.getEnabledFeatureIds().isEmpty()) {
```

Comment:
Would there be cases when detector is already running here and then we see features are disbaled and then we throw the endRunException? Would the features be disabled in this case cause of some other error? or this is incase features were never enabled in the first place?

Reply:
When detector is running, we cannot edit detector so cannot disable features. Disabling is done before starting a detector.